app/api/responses.js

```diff
@@ -0,0 +1,137 @@
+    const enrollmentStudent = await db.Enrollment.findOne({
+        where: { role: 'student', userId: user.id },
+        include: [
+            {
+                model: db.Section,
+                required: true,
+                where: { courseId: courseId }
+            }
+        ]
+    })
+
+    // check to make sure response exists and belongs to the user making the request
+    const oldResponse = await db.Response.findOne({
+        where: { id: responseId },
+        include: [
+            {
+                model: db.Enrollment,
+                required: true,
+                where: { userId: user.id }
+            }
+        ]
+    })
+
+    if (enrollmentStudent) {
+        // for now, only multiple choice and multiple answer available
+        if (req.body.answers && Object.keys(req.body.answers).length > 1) { // request must have an answer and it must have at least two options to pick from 
+            //const oldResponse = await db.Response.findByPk(responseId)
+            if (oldResponse) {
+                try {
+                    const questionInLecture = await questionService.getQuestionInLecture(questionId, lectureId)
+                    if (questionInLecture) { // if we found a valid questionInLecture
+                        if (questionInLecture.published === true) {
+                            // if the question in lecture exists, we know there is a question that we can check for scoring
+                            const question = await questionService.getQuestionInCourse(questionId, courseId)
+                            const questionScore = questionService.getQuestionScore(question, req.body)
+                            const responseToUpdate = {
+                                score: questionScore,
+                                submission: req.body.answers
+                            }
+                            const response = await oldResponse.update(responseToUpdate)
+                            res.status(200).send({
+                                response: responseService.extractResponseFields(response)
+                            })
+                        } else { // question is no longer available
+                            res.status(400).send({ error: `The question is not published` })
+                        }
+                    } else {
+                        res.status(404).send({error: `No question in lecture found for the given lecture and question`})
+                    }
+                } catch (e) {
+                    next(e)
+                }
+            } else {
+                // maybe better to have a different status code if it belongs to another user
+                // the thought here is that if it does belong to another user it would be better to not explicitly tell an attacker that
+                res.status(404).send({ error: 'response with given id either not found or belongs to another user' })
```

Comment:
I would just say not found. I don't think there is a need to explain that it could belong to another user